packages/storage/src/platform/node/connection.ts

```diff
@@ -120,12 +115,88 @@ export class FetchConnection implements Connection {
   }
 }
 
-export function newConnection(): Connection {
-  return connectionFactoryOverride
-    ? connectionFactoryOverride()
-    : new FetchConnection();
+export class FetchTextConnection extends FetchConnection<string> {
+  getResponse(): string {
+    if (!this.body_) {
+      throw internalError('cannot .getResponse() before receiving response');
+    }
+    return Buffer.from(this.body_).toString('utf-8');
+  }
+}
+
+export function newTextConnection(): Connection<string> {
+  return textFactoryOverride
+    ? textFactoryOverride()
+    : new FetchTextConnection();
+}
+
+export class FetchBytesConnection extends FetchConnection<ArrayBuffer> {
+  getResponse(): ArrayBuffer {
+    if (!this.body_) {
+      throw internalError('cannot .getResponse() before sending');
+    }
+    return this.body_;
+  }
+}
+
+export function newBytesConnection(): Connection<ArrayBuffer> {
+  return new FetchBytesConnection();
+}
+
+export class FetchStreamConnection extends FetchConnection<NodeJS.ReadableStream> {
+  private stream_: NodeJS.ReadableStream | null = null;
+
+  async send(
+    url: string,
+    method: string,
+    body?: ArrayBufferView | Blob | string,
+    headers?: Record<string, string>
+  ): Promise<void> {
+    if (this.sent_) {
+      throw internalError('cannot .send() more than once');
+    }
+    this.sent_ = true;
+
+    try {
+      const response = await nodeFetch(url, {
```

Comment:
```suggestion
      const response = await this.fetch_(url, {
```

so you can stub fetch easily in tests.